geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert/Transformers.scala

```diff
@@ -117,16 +146,24 @@ object Transformers extends JavaTokenParsers {
   case class LitString(value: String) extends Lit[String]
   case class LitInt(value: Integer) extends Lit[Integer]
   case class LitLong(value: Long) extends Lit[Long]
+  case class LitFloat(value: java.lang.Float) extends Lit[java.lang.Float]
   case class LitDouble(value: java.lang.Double) extends Lit[java.lang.Double]
+  case class LitBoolean(value: java.lang.Boolean) extends Lit[java.lang.Boolean]
+
   case class Cast2Int(e: Expr) extends Expr {
     override def eval(args: Array[Any])(implicit ctx: EvaluationContext): Any = e.eval(args).asInstanceOf[String].toInt
   }
-
+  case class Cast2Long(e: Expr) extends Expr {
+    override def eval(args: Array[Any])(implicit ctx: EvaluationContext): Any = e.eval(args).asInstanceOf[String].toLong
```

Comment:
I'm getting this issue now with empty strings using the xml parser:
```
{ name = elevation, path = "ele/text()", transform = "$0::double" }
```
```
java.lang.NumberFormatException: empty String
	at sun.misc.FloatingDecimal.readJavaFormatString(FloatingDecimal.java:1020)
	at java.lang.Double.parseDouble(Double.java:540)
	at scala.collection.immutable.StringLike$class.toDouble(StringLike.scala:232)
	at scala.collection.immutable.StringOps.toDouble(StringOps.scala:31)
	at org.locationtech.geomesa.convert.Transformers$Cast2Double.eval(Transformers.scala:126)

```

This goes back to the null checks, but slightly different since it's coming from xml. So something that can handle empty strings is what I need here.